src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexPersistence.cs

```diff
@@ -111,10 +113,59 @@ public override void Clean(IndexCleanup mode)
         // lucene method
     }
 
+    private const int IndexingCompressionMaxTestDocuments = 10000;
+
+
     public override void Initialize(StorageEnvironment environment)
     {
     }
 
+    public override void OnInitializeComplete()
+    {
+        var contextPool = _index._contextPool;
+
+        using (contextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (var tx = context.OpenReadTransaction())
+        {
+            // It is already initialized, we can return.
+            if (CompactTree.HasDictionary(tx.InnerTransaction.LowLevelTransaction))
+                return;
+        }
+
+        if (_index.SourceType != IndexSourceType.Documents)
+            return;
+
+        var documentStorage = _index.DocumentDatabase.DocumentsStorage;
+
+        using var _ = CultureHelper.EnsureInvariantCulture();
+        using var __ = contextPool.AllocateOperationContext(out TransactionOperationContext indexContext);
+        using var queryContext = QueryOperationContext.Allocate(_index.DocumentDatabase, _index);
+
+        using (CurrentIndexingScope.Current = _index.CreateIndexingScope(indexContext, queryContext))
+        {
+            indexContext.PersistentContext.LongLivedTransactions = true;
+            queryContext.SetLongLivedTransactions(true);
+            queryContext.OpenReadTransaction();
+
+            using var tx = indexContext.OpenWriteTransaction();
+
+            // We are creating a new converter because converters get tied through their accessors to the structure, and since on Map-Reduce indexes
+            // we only care about the map and not the reduce hilarity can ensure when properties do not share the type. 
+            var converter = CreateConverter(_index);
+            converter.IgnoreComplexObjectsDuringIndex = true; // for training, we don't care
+            var enumerator = new CoraxDocumentTrainEnumerator(indexContext, converter, _index, _index.Type, documentStorage, queryContext, _index.Collections, _index.Configuration.DocumentsLimitForCompressionDictionaryCreation);
+            var testEnumerator = new CoraxDocumentTrainEnumerator(indexContext, converter, _index, _index.Type, documentStorage, queryContext, _index.Collections, IndexingCompressionMaxTestDocuments);
```

Comment:
IndexingCompressionMaxTestDocumen comes from a const, should this be configurable or I do not know e.g `_index.Configuration.DocumentsLimitForCompressionDictionaryCreation / 10` ?

Reply:
@redknightlois ?